### PR TITLE
fix: issue #112

### PR DESCRIPTION
**Pull request by yoution**

### CODE REVIEW

**Comment by yoution** [comment]
@maxceem please review

File: src/hoc/withAuthentication/index.js [hunk 1]
```diff
@@ -96,8 +105,14 @@ export default function withAuthentication(Component) {
             <LoadingIndicator error={authError || teamMembersLoadingError} />
           ))}
 
+        {/* Show component only if  v5 user profile load error */}
+        {isLoggedIn === true && v5UserProfileLoadingError && (
+          <LoadingIndicator error={"Error: Network Error"} />
```
Comment (maxceem) [inline src/hoc/withAuthentication/index.js:110]: @yoution sorry, I didn't mean to show error `Network Error` I mean to show error which happens during loading user profile. We have to pass it there.

File: src/hoc/withAuthentication/actions/index.js [hunk 2]
```diff
@@ -40,6 +40,19 @@ export const authLoadTeamMembers = (teamId) => ({
   },
 });
 
+/**
+ * Loads v5 user profile for authentication/permission purposes
+ *
+ * @returns {Promise} loaded user profile
+ */
+export const authLoadUserProfile = (teamId) => ({
```
Comment (maxceem) [inline src/hoc/withAuthentication/actions/index.js:48]: argument `teamId` is not used

**Review by maxceem — CHANGES_REQUESTED** [review]
@yoution strange, but for me it doesn't work. Does it work for you?

image

**Comment by maxceem** [comment]
@yoution also, could we please name action as `AUTH_LOAD_V5_USER_PROFILE` instead of `AUTH_LOAD_USER_PROFILE` to keep it clear. Because we also have another user profile (v3).

File: src/hoc/withAuthentication/index.js (hunk 1 above)
Comment (yoution) [reply src/hoc/withAuthentication/index.js:110]: I don't know how to repreduce the error case

Comment (yoution) [reply src/hoc/withAuthentication/index.js:110]: image
it works for me

Comment (maxceem) [reply src/hoc/withAuthentication/index.js:110]: you may change endpoint to incorrect one, like below, it should cause some error during request, and it should be shown on the page
```js
return axios.get(`${config.API.V5}/taas-teams/meBROKEN`);
```

Comment (yoution) [reply src/hoc/withAuthentication/index.js]: image
how about this

Comment (maxceem) [reply src/hoc/withAuthentication/index.js]: yeah, looks good, this is what was needed

**Review by maxceem — CHANGES_REQUESTED** [review]
@yoution it works almost perfect for me locally.

But there is a small improvement to make.
When we wait for v5 user profile loading, the loading indicator has to be shown

image

It's hard to see, but when V5 user profile is being loaded, the loading indicator is not shown, see how it looks if I slow down loading of v5 user profile to 10 seconds https://monosnap.com/file/CuX6Ahly89FgAdpn1QJFRgh7OcJRF9

How to reproduce it. In file `src/services/teams.js`
- add import `import { delay } from "utils/helpers";` on line 6
- update code of `getUserProfile` to
  ```js
    export const getUserProfile = () => {
      return delay(10000).then(() => axios.get(`${config.API.V5}/taas-teams/me`));
    };  
  ```
  
So you could see how it would look like if loading of V5 user profile is slow.

It would be nice (but not required) if existent code could be reused, we already have 2 places in code: 1 place to show loading indicator or error, and another one to show component if all is loaded:

image

One more thing.
Could we please also rename 
- action method `authLoadUserProfile` to `authLoadV5UserProfile`
- and service method `getUserProfile` to `getV5UserProfile`

to make it clear that this is for V5 user profile to avoid confusion in the future.

**Review by maxceem — APPROVED** [review]
@yoution thanks for the quick updates. Works good now.

**Pull request merged**